backend/main.py

```diff
@@ -1,57 +1,84 @@
+# Backend API for Art Trend Classifier
 import os
 import io
 import torch
 import torch.nn as nn
 from torchvision import models, transforms
 from PIL import Image
 import uvicorn
-from fastapi import FastAPI, File, UploadFile, HTTPException
+from fastapi import FastAPI, File, UploadFile, HTTPException, Request
 from fastapi.middleware.cors import CORSMiddleware
 from typing import Dict, Any
 import albumentations as A
 from albumentations.pytorch import ToTensorV2
-import cv2
 import numpy as np
+import time
 
 from contextlib import asynccontextmanager
+from slowapi import Limiter, _rate_limit_exceeded_handler
+from slowapi.util import get_remote_address
+from slowapi.errors import RateLimitExceeded
+
+# Import our refactored modules
+from config import IMAGE_LIMITS, SECURITY_CONFIG, API_CONFIG
+from validation import validate_file_input, process_image_from_content, ValidationError
+from reviews import generate_review
+from logging_utils import setup_logging, AnalysisLogger
+
+# Setup logging
+logger = setup_logging(is_lambda=False)
 
 @asynccontextmanager
 async def lifespan(app: FastAPI):
-    # Startup
+    # Load model on startup
     try:
         load_model()
+        print(f"Model loaded in lifespan: {model is not None}")
+        print(f"Class names loaded in lifespan: {len(class_names)}")
     except Exception as e:
         print(f"Error loading model: {e}")
         raise
     yield
-    # Shutdown
     print("Shutting down...")
 
 app = FastAPI(title="Art Trend Classifier", version="1.0.0", lifespan=lifespan)
 
-# Enable CORS for frontend
+# Security headers middleware
+@app.middleware("http")
+async def add_security_headers(request: Request, call_next):
+    response = await call_next(request)
+    for header, value in SECURITY_CONFIG.HEADERS.items():
+        response.headers[header] = value
+    return response
+
+# Rate limiting configuration
+limiter = Limiter(key_func=get_remote_address, default_limits=[API_CONFIG.RATE_LIMITS["default"]])
+app.state.limiter = limiter
+app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)
+
+# Enable CORS for frontend - RESTRICTED origins only
 app.add_middleware(
     CORSMiddleware,
-    allow_origins=["*"],
-    allow_credentials=True,
-    allow_methods=["*"],
-    allow_headers=["*"],
+    allow_origins=API_CONFIG.CORS_ORIGINS,
+    allow_credentials=False,  # Disabled for security
+    allow_methods=["GET", "POST", "OPTIONS"],
+    allow_headers=["Content-Type", "Authorization"],
 )
 
-# Global variables for model
+# Global model state
 model = None
 class_names = []
 device = torch.device('cuda' if torch.cuda.is_available() else (
     'mps' if hasattr(torch.backends, 'mps') and torch.backends.mps.is_available() else 'cpu'))
 
 def load_model():
-    # Load the trained model
+    """Load ResNet50 model with custom classifier head"""
     global model, class_names
```

Comment:
You should avoid `global` things and correctly utilize dependecy injection pattern